v0/destinations/klaviyo/transform.js

```diff
@@ -160,13 +161,75 @@ const trackRequestHandler = (message, category, destination) => {
     WhiteListedTraits
   );
   customerProperties = removeUndefinedAndNullValues(customerProperties);
-  if (destination.Config.enforceEmailAsPrimary) {
-    delete customerProperties.$id;
-    customerProperties._id = getFieldValueFromMessage(message, "userId");
-  }
-  payload.customer_properties = customerProperties;
-  if (message.timestamp) {
-    payload.time = toUnixTimestamp(message.timestamp);
+  return customerProperties;
+};
+const trackRequestHandler = (message, category, destination) => {
+  let payload = {};
+  let event = getValueFromMessage(message, "event");
+  event = event ? event.trim().toLowerCase() : event;
+  if (ecomEvents.includes(event) && message.properties) {
+    const eventName = eventNameMapping[event];
+    payload.event = eventName;
+    payload.token = destination.Config.publicApiKey;
+    const eventMap = jsonNameMapping[eventName];
+    payload.customer_properties = createCustomerProperties(message);
+    if (
+      !payload.customer_properties.$email &&
+      !payload.customer_properties.$phone_number
+    ) {
+      throw new CustomError("customer_properties not found", 400);
+    }
+    const categ = CONFIG_CATEGORIES[eventMap];
+    payload.properties = constructPayload(
+      message.properties,
+      MAPPING_CONFIG[categ.name]
+    );
+    if (message.properties.items) {
+      const itemArr = [];
+      message.properties.items.forEach(key => {
+        let item = constructPayload(
+          key,
+          MAPPING_CONFIG[CONFIG_CATEGORIES.ITEMS.name]
+        );
+        item = removeUndefinedAndNullValues(item);
+        itemArr.push(item);
+      });
+      if (!payload.properties) {
+        payload.properties = {};
+      }
+      payload.properties.items = itemArr;
+    }
+    if (isEmptyObject(payload.properties)) {
+      let properties = {};
+      properties = extractCustomFields(
+        message,
+        properties,
+        ["properties"],
+        ecomExclusionKeys
+      );
+      if (!isEmptyObject(properties)) {
+        payload = {
+          ...payload,
+          properties
+        };
+      }
+    }
+  } else {
+    payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+    payload.token = destination.Config.publicApiKey;
+    if (message.properties && message.properties.revenue) {
+      payload.properties.$value = message.properties.revenue;
+      delete payload.properties.revenue;
+    }
+    const customerProperties = createCustomerProperties(message);
+    if (destination.Config.enforceEmailAsPrimary) {
+      delete customerProperties.$id;
+      customerProperties._id = getFieldValueFromMessage(message, "userId");
+    }
+    payload.customer_properties = customerProperties;
+    if (message.timestamp) {
+      payload.time = toUnixTimestamp(message.timestamp);
+    }
```

Comment:
The entire logic is cramped here with no visual separation or code documentation. Please improve this.

Reply:
have added comments and separated different things with linebreak